xml/upgrade-3.10-4.xsl

```diff
@@ -116,16 +116,38 @@
 
 <!-- Drop groups that would become empty -->
 <xsl:template match="group">
-    <xsl:if test="count(.|$dropped_groups) != count($dropped_groups)">
-        <xsl:call-template name="identity"/>
-    </xsl:if>
+    <xsl:choose>
+        <xsl:when test="count(.|$dropped_groups) = count($dropped_groups)">
+            <xsl:call-template name="info">
+                <xsl:with-param name="msg"
+                                select="concat('Dropping group ', @id,
+                                               ' because it would become',
+                                               ' empty')"/>
```

Comment:
This info does make sense, since it's a ripple effect of a behavioral change

Reply:
Also if an ACL refers to a particular group or clone, then as of right now removing the group/clone would cause a schema validation error.

We may also want to keep the "split top-level rules" for the same reason.

Others cases are more questionable; see https://github.com/ClusterLabs/pacemaker/pull/3741#discussion_r1868520545.